controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -696,49 +696,65 @@ public void listStreamsInScope(Controller.StreamsInScopeRequest request, StreamO
+                listStreamsInternal(scopeName, request.getContinuationToken().getToken(), pageLimit, ctx)
                         .handle((response, ex) -> {
                             if (ex != null) {
                                 if (Exceptions.unwrap(ex) instanceof StoreException.DataNotFoundException) {
-                                    return Controller.StreamsInScopeResponse.newBuilder().setStatus(Controller.StreamsInScopeResponse.Status.SCOPE_NOT_FOUND).build();
+                                    return Controller.StreamsInScopeResponse
+                                            .newBuilder().setStatus(Controller.StreamsInScopeResponse.Status.SCOPE_NOT_FOUND).build();
                                 } else {
                                     throw new CompletionException(ex);
                                 }
                             } else {
-                                log.debug("All streams in scope with continuation token: {}", response);
-                                List<StreamInfo> streams = response
-                                        .getKey().stream()
-                                        .filter(streamName -> {
-                                            String streamAuthResource =
-                                                    AuthResourceRepresentation.ofStreamInScope(scopeName, streamName);
-
-                                            boolean isAuthorized = grpcAuthHelper.isAuthorized(streamAuthResource,
-                                                    AuthHandler.Permissions.READ, ctx);
-                                            log.debug("Authorization for [{}] for READ permission was [{}]",
-                                                    streamAuthResource, isAuthorized);
-                                            return isAuthorized;
-                                        })
-                                        .map(m -> StreamInfo.newBuilder().setScope(scopeName).setStream(m).build())
-                                        .collect(Collectors.toList());
                                 return Controller.StreamsInScopeResponse
-                                        .newBuilder().addAllStreams(streams)
-                                        .setContinuationToken(Controller.ContinuationToken.newBuilder()
-                                                                .setToken(response.getValue()).build())
+                                        .newBuilder().addAllStreams(response.getKey())
+                                        .setContinuationToken(Controller.ContinuationToken.newBuilder().setToken(response.getValue()).build())
                                         .setStatus(Controller.StreamsInScopeResponse.Status.SUCCESS).build();
                             }
-                        }), responseObserver, requestTag);
+                        });
+
+        authenticateExecuteAndProcessResults(
```

Comment:
Passing streamsFn as an argument to authenticateExecuteAndProcessResults is a much cleaner way to organize code. Could we also do the same for listKeyValueTablesInScope?